streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -870,43 +900,75 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
                 cacheSizePerThread,
                 stateDirectory,
                 delegatingStateRestoreListener,
-                i + 1,
+                threadIdx,
                 KafkaStreams.this::closeToError,
-                this::defaultStreamsUncaughtExceptionHandler
-            );
-            threads.add(streamThread);
-            threadState.put(streamThread.getId(), streamThread.state());
-            storeProviders.add(new StreamThreadStateStoreProvider(streamThread));
-        }
+                streamsUncaughtExceptionHandler
```

Comment:
nit: If it happens that you need to push another commit, could you fix the indentation here? Sorry that I haven't noticed this before.

Reply:
ah good catch. the diff makes that hard to see as it was actually moved to a new method.